### PR TITLE
pd: add configuration for store_limit_v2

### PR DESCRIPTION
**Pull request by bufferflies**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

In v7.1.0, PD introduces `store-limit-version` to control the store limit formula version.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: pd-control.md [hunk 1]
```diff
@@ -451,7 +451,9 @@ config show cluster-version
 
 - `enable-placement-rules` 用于开启 placement rules，在 v5.0 及以上的版本默认开启。
 
-- `store-limit-mode` 用于控制 store 限速机制的模式。主要有两种模式：`auto` 和 `manual`。`auto` 模式下会根据 load 自动进行平衡调整（实验性功能）。
+- `store-limit-mode` 用于控制 store 限速机制的模式。主要有两种模式：`auto` 和 `manual`。`auto` 模式下会根据 load 自动进行平衡调整（弃用）。
+
+- `store-limit-version` 用于控制 `store limit` 限制模式，主要有两种方式 `v1` 和 `v2`。 默认值为 `v1`，通过手动修改 `store limit`，来限制单个 TIKV 调度速度。在 `v2` 模式下，用户将不需要关注 `store limit` 值，PD 会根据 TIKV Snapshot 执行情况进行动态调整调度速度（实验特性）。
```
Comment (Oreoxmt) [inline pd-control.md:456]:
```suggestion
- `store-limit-version` 用于设置 `store limit` 限制模式，目前提供两种方式：`v1` 和 `v2`。默认值为 `v1`，你可以手动修改 `store limit` 以限制单个 TiKV 调度速度。`v2` 模式为实验特性，在 `v2` 模式下，你无需关注 `store limit` 值，PD 将根据 TiKV Snapshot 执行情况动态调整 TiKV 调度速度。详情请参考 [store limit v2 原理](/configure-store-limit.md#store-limit-v2-原理)。
```

File: configure-store-limit.md [hunk 2]
```diff
@@ -67,3 +67,9 @@ store limit 1 5                     // 设置 store 1 添加和删除 peer 的
 store limit 1 5 add-peer            // 设置 store 1 添加 peer 的速度上限为每分钟 5 个。
 store limit 1 5 remove-peer         // 设置 store 1 删除 peer 的速度上限为每分钟 5 个。
 ```
+
+
+## Store Limit V2 原理
+当 `store-limit-version` 设置成 `v2` 时生效，operator 调度限制将会根据 TIKV Snapshot 执行情况进行动态调整。当 TIKV 积压的任务越少，PD 将会增加其调度任务。反之，PD 将会减少对该节点的调度任务。用户不再需要关注如何设置 `store limit` 来加速迁移进度。
+
+在该模式下，TIKV 执行速度成为迁移进度的最大瓶颈，用户可以根据 TIKV DeTails --> Snapshot --> Snapshot Speed 面板来判断当前调度速度是否达到 TIKV 限流设置。通过调整 TIKV Snapshot Limit (`snap_io_max_bytes_per_sec`) 来增加/减少该节点的调度速度。
```
Comment (Oreoxmt) [inline configure-store-limit.md:75]:
```suggestion
## Store Limit v2 原理

当 [`store-limit-version`](/pd-configuration-file.md#store-limit-version-从-v710-版本开始引入) 设置为 `v2` 时，Store Limit v2 生效。在此模式下，Operator 调度限制将根据 TiKV Snapshot 执行情况进行动态调整。当 TiKV 积压的任务较少时，PD 会增加其调度任务。相反，PD 会减少对该节点的调度任务。此时，你无需关注如何设置 `store limit` 以加速调度进度。

在该模式下，TiKV 执行速度成为迁移进度的主要瓶颈。你可以通过 **TiKV DeTails** > **Snapshot** > **Snapshot Speed** 面板判断当前调度速度是否达到 TiKV 限流设置。通过调整 TiKV Snapshot Limit ([`snap-io-max-bytes-per-sec`](/tikv-configuration-file.md#snap-io-max-bytes-per-sec) 来增加或减少该节点的调度速度。
```

Comment (bufferflies) [reply configure-store-limit.md]: fixed

File: pd-control.md (hunk 1 above)
Comment (bufferflies) [reply pd-control.md]: fixed

File: pd-configuration-file.md [hunk 3]
```diff
@@ -403,6 +403,12 @@ pd-server 相关配置项。
 + 默认值：true
 + 参考 [Placement Rules 使用文档](/configure-placement-rules.md)
 
+
+### `store-limit-version`
++ 设置 store limit 工作模式
++ 默认值：v1
+
+
```
[the remark this answers is not in the document]
Comment (bufferflies) [reply pd-configuration-file.md]: fixed

File: configure-store-limit.md [hunk 4]
```diff
@@ -67,3 +67,9 @@ store limit 1 5                     // 设置 store 1 添加和删除 peer 的
 store limit 1 5 add-peer            // 设置 store 1 添加 peer 的速度上限为每分钟 5 个。
 store limit 1 5 remove-peer         // 设置 store 1 删除 peer 的速度上限为每分钟 5 个。
 ```
+
+
+## Store Limit V2 原理
+
+当 [`store-limit-version`](/pd-configuration-file.md#store-limit-version-从-v710-版本开始引入) 设置为 `v2` 时，Store Limit v2 生效。在此模式下，Operator 调度限制将根据 TiKV Snapshot 执行情况进行动态调整。当 TiKV 积压的任务较少时，PD 会增加其调度任务。相反，PD 会减少对该节点的调度任务。此时，你无需关注如何设置 `store limit` 以加速调度进度。
+在该模式下，TiKV 执行速度成为迁移进度的主要瓶颈。你可以通过 **TiKV DeTails** > **Snapshot** > **Snapshot Speed** 面板判断当前调度速度是否达到 TiKV 限流设置。通过调整 TiKV Snapshot Limit ([`snap-io-max-bytes-per-sec`](/tikv-configuration-file.md#snap-io-max-bytes-per-sec) 来增加或减少该节点的调度速度。
```
Comment (Oreoxmt) [inline configure-store-limit.md:75]:
```suggestion

## Store Limit v2 原理

当 [`store-limit-version`](/pd-configuration-file.md#store-limit-version-从-v710-版本开始引入) 设置为 `v2` 时，Store Limit v2 生效。在此模式下，Operator 调度限制将根据 TiKV Snapshot 执行情况进行动态调整。当 TiKV 积压的任务较少时，PD 会增加其调度任务。相反，PD 会减少对该节点的调度任务。此时，你无需关注如何设置 `store limit` 以加速调度进度。

在该模式下，TiKV 执行速度成为迁移进度的主要瓶颈。你可以通过 **TiKV DeTails** > **Snapshot** > **Snapshot Speed** 面板判断当前调度速度是否达到 TiKV 限流设置。通过调整 TiKV Snapshot Limit ([`snap-io-max-bytes-per-sec`](/tikv-configuration-file.md#snap-io-max-bytes-per-sec) 来增加或减少该节点的调度速度。
```

File: pd-configuration-file.md [hunk 5]
```diff
@@ -403,6 +403,19 @@ pd-server 相关配置项。
 + 默认值：true
 + 参考 [Placement Rules 使用文档](/configure-placement-rules.md)
 
+
+### `store-limit-version`
```
Comment (Oreoxmt) [inline pd-configuration-file.md:407]:
```suggestion
### `store-limit-version` <span class="version-mark">从 v7.1.0 版本开始引入</span>

```

File: pd-configuration-file.md [hunk 6]
```diff
@@ -403,6 +403,19 @@ pd-server 相关配置项。
...
+>
+> 在当前版本中，将该配置项设置为 `"v2"` 为实验特性，不建议在生产环境中使用。
+
++ 设置 store limit 工作模式
++ 默认值：v1
++ 可选值：
+    + v1：在 v1 模式下，你可以手动修改 `store limit` 以限制单个 TiKV 调度速度。
+    + v2：（实验特性）在 v2 模式下，你无需关注 `store limit` 值，PD 将根据 TiKV Snapshot 执行情况动态调整 TiKV 调度速度。详情请参考 [store limit v2 原理](/configure-store-limit.md#store-limit-v2-原理)。
+
+
```
Comment (Oreoxmt) [inline pd-configuration-file.md:418]:
```suggestion

```

File: pd-configuration-file.md [hunk 7]
```diff
@@ -403,6 +403,18 @@ pd-server 相关配置项。
...
+> **警告：**
+>
+> 在当前版本中，将该配置项设置为 `"v2"` 为实验特性，不建议在生产环境中使用。
+
++ 设置 store limit 工作模式
++ 默认值：v1
++ 可选值：
+    + v1：在 v1 模式下，你可以手动修改 `store limit` 以限制单个 TiKV 调度速度。
+    + v2：（实验特性）在 v2 模式下，你无需关注 `store limit` 值，PD 将根据 TiKV Snapshot 执行情况动态调整 TiKV 调度速度。详情请参考 [store limit v2 原理](/configure-store-limit.md#store-limit-v2-原理)。
```
Comment (Oreoxmt) [inline pd-configuration-file.md:417]:
```suggestion
    + v2：（实验特性）在 v2 模式下，你无需关注 `store limit` 值，PD 将根据 TiKV Snapshot 执行情况动态调整 TiKV 调度速度。详情请参考 [store limit v2 原理](/configure-store-limit.md#store-limit-v2-原理)。

```

File: pd-configuration-file.md [hunk 8]
```diff
@@ -403,6 +403,19 @@ pd-server 相关配置项。
 + 默认值：true
 + 参考 [Placement Rules 使用文档](/configure-placement-rules.md)
 
+
+### `store-limit-version` <span class="version-mark">从 v7.1.0 版本开始引入</span>
```
Comment (Oreoxmt) [inline pd-configuration-file.md:407]:
```suggestion
### `store-limit-version` <span class="version-mark">从 v7.1.0 版本开始引入</span>
```

File: configure-store-limit.md [hunk 9]
```diff
@@ -67,3 +67,9 @@ store limit 1 5                     // 设置 store 1 添加和删除 peer 的
 store limit 1 5 add-peer            // 设置 store 1 添加 peer 的速度上限为每分钟 5 个。
 store limit 1 5 remove-peer         // 设置 store 1 删除 peer 的速度上限为每分钟 5 个。
 ```
+
+## Store Limit v2 原理
+
+当 [`store-limit-version`](/pd-configuration-file.md#store-limit-version-从-v710-版本开始引入) 设置为 `v2` 时，Store Limit v2 生效。在此模式下，Operator 调度限制将根据 TiKV Snapshot 执行情况进行动态调整。当 TiKV 积压的任务较少时，PD 会增加其调度任务。相反，PD 会减少对该节点的调度任务。此时，你无需关注如何设置 `store limit` 以加速调度进度。
```
Comment (hfxsd) [inline configure-store-limit.md:73]:
```suggestion
当 [`store-limit-version`](/pd-configuration-file.md#store-limit-version-从-v710-版本开始引入) 设置为 `v2` 时，Store Limit v2 生效。在此模式下，Operator 调度限制将根据 TiKV Snapshot 执行情况进行动态调整。当 TiKV 积压的任务较少时，PD 会增加其调度任务。相反，PD 会减少对该节点的调度任务。此时，你无需关注如何设置 `store limit` 以加快调度进度。
```

File: pd-configuration-file.md [hunk 10]
```diff
@@ -403,6 +403,18 @@ pd-server 相关配置项。
 + 默认值：true
 + 参考 [Placement Rules 使用文档](/configure-placement-rules.md)
 
+### `store-limit-version` <span class="version-mark">从 v7.1.0 版本开始引入</span>
+
+> **警告：**
+>
+> 在当前版本中，将该配置项设置为 `"v2"` 为实验特性，不建议在生产环境中使用。
+
++ 设置 store limit 工作模式
```
Comment (hfxsd) [inline pd-configuration-file.md:412]:
```suggestion
+ 设置 `store limit` 工作模式
```

File: configure-store-limit.md [hunk 11]
```diff
@@ -67,3 +67,9 @@ store limit 1 5                     // 设置 store 1 添加和删除 peer 的
...
+
+在该模式下，TiKV 执行速度成为迁移进度的主要瓶颈。你可以通过 **TiKV DeTails** > **Snapshot** > **Snapshot Speed** 面板判断当前调度速度是否达到 TiKV 限流设置。通过调整 TiKV Snapshot Limit ([`snap-io-max-bytes-per-sec`](/tikv-configuration-file.md#snap-io-max-bytes-per-sec) 来增加或减少该节点的调度速度。
```
Comment (hfxsd) [inline configure-store-limit.md:75]:
```suggestion
在该模式下，TiKV 执行速度成为迁移进度的主要瓶颈。你可以通过 **TiKV DeTails** > **Snapshot** > **Snapshot Speed** 面板判断当前调度速度是否达到 TiKV 限流设置。通过调整 TiKV Snapshot Limit ([`snap-io-max-bytes-per-sec`](/tikv-configuration-file.md#snap-io-max-bytes-per-sec)) 来增加或减少该节点的调度速度。
```

File: pd-control.md [hunk 12]
```diff
@@ -451,7 +451,9 @@ config show cluster-version
 
 - `enable-placement-rules` 用于开启 placement rules，在 v5.0 及以上的版本默认开启。
 
-- `store-limit-mode` 用于控制 store 限速机制的模式。主要有两种模式：`auto` 和 `manual`。`auto` 模式下会根据 load 自动进行平衡调整（实验性功能）。
+- `store-limit-mode` 用于控制 store 限速机制的模式。主要有两种模式：`auto` 和 `manual`。`auto` 模式下会根据 load 自动进行平衡调整（弃用）。
+
+- `store-limit-version` 用于设置 `store limit` 限制模式，目前提供两种方式：`v1` 和 `v2`。默认值为 `v1`，你可以手动修改 `store limit` 以限制单个 TiKV 调度速度。`v2` 模式为实验特性，在 `v2` 模式下，你无需关注 `store limit` 值，PD 将根据 TiKV Snapshot 执行情况动态调整 TiKV 调度速度。详情请参考 [store limit v2 原理](/configure-store-limit.md#store-limit-v2-原理)。
```
Comment (hfxsd) [inline pd-control.md:456]:
```suggestion
- `store-limit-version` 用于设置 `store limit` 限制模式，目前提供两种方式：`v1` 和 `v2`。默认值为 `v1`，你可以手动修改 `store limit` 以限制单个 TiKV 调度速度。`v2` 模式为实验特性，在 `v2` 模式下，你无需关注 `store limit` 值，PD 将根据 TiKV Snapshot 执行情况动态调整 TiKV 调度速度。详情请参考 [Store Limit v2 原理](/configure-store-limit.md#store-limit-v2-原理)。
```

**Review by hfxsd — APPROVED** [review]
Rest LGTM

**Review by tonyxuqqi — APPROVED** [review]

**Review by benmaoer — APPROVED** [review]
LGTM

File: pd-control.md [hunk 13]
```diff
@@ -451,7 +451,9 @@ config show cluster-version
 
 - `enable-placement-rules` 用于开启 placement rules，在 v5.0 及以上的版本默认开启。
 
-- `store-limit-mode` 用于控制 store 限速机制的模式。主要有两种模式：`auto` 和 `manual`。`auto` 模式下会根据 load 自动进行平衡调整（实验性功能）。
+- `store-limit-mode` 用于控制 store 限速机制的模式。主要有两种模式：`auto` 和 `manual`。`auto` 模式下会根据 load 自动进行平衡调整（弃用）。
+
+- `store-limit-version` 用于设置 `store limit` 限制模式，目前提供两种方式：`v1` 和 `v2`。默认值为 `v1`。在 `v1` 模式下，你可以手动修改 `store limit` 以限制单个 TiKV 调度速度。`v2` 模式为实验特性，在 `v2` 模式下，你无需关注 `store limit` 值，PD 将根据 TiKV Snapshot 执行情况动态调整 TiKV 调度速度。详情请参考 [Store Limit v2 原理](/configure-store-limit.md#store-limit-v2-原理)。
```
Comment (Oreoxmt) [inline pd-control.md:456]:
````suggestion
- `store-limit-version` 用于设置 `store limit` 限制模式，目前提供两种方式：`v1` 和 `v2`。默认值为 `v1`。在 `v1` 模式下，你可以手动修改 `store limit` 以限制单个 TiKV 调度速度。`v2` 模式为实验特性，在 `v2` 模式下，你无需关注 `store limit` 值，PD 将根据 TiKV Snapshot 执行情况动态调整 TiKV 调度速度。详情请参考 [Store Limit v2 原理](/configure-store-limit.md#store-limit-v2-原理)。

    ```bash
    config set store-limit-version v2       // 使用 Store Limit v2
    ```
````

File: configure-store-limit.md [hunk 14]
```diff
@@ -67,3 +67,9 @@ store limit 1 5                     // 设置 store 1 添加和删除 peer 的
 store limit 1 5 add-peer            // 设置 store 1 添加 peer 的速度上限为每分钟 5 个。
 store limit 1 5 remove-peer         // 设置 store 1 删除 peer 的速度上限为每分钟 5 个。
 ```
+
+## Store Limit v2 原理
+
+当 [`store-limit-version`](/pd-configuration-file.md#store-limit-version-从-v710-版本开始引入) 设置为 `v2` 时，Store Limit v2 生效。在此模式下，Operator 调度限制将根据 TiKV Snapshot 执行情况进行动态调整。当 TiKV 积压的任务较少时，PD 会增加其调度任务。相反，PD 会减少对该节点的调度任务。此时，你无需关注如何设置 `store limit` 以加快调度进度。
+
+在该模式下，TiKV 执行速度成为迁移进度的主要瓶颈。你可以通过 **TiKV DeTails** > **Snapshot** > **Snapshot Speed** 面板判断当前调度速度是否达到 TiKV 限流设置。通过调整 TiKV Snapshot Limit ([`snap-io-max-bytes-per-sec`](/tikv-configuration-file.md#snap-io-max-bytes-per-sec)) 来增加或减少该节点的调度速度。
```
Comment (ran-huang) [inline configure-store-limit.md:75]: 这里是写错了还是就这样的...
```suggestion
在该模式下，TiKV 执行速度成为迁移进度的主要瓶颈。你可以通过 **TiKV Details** > **Snapshot** > **Snapshot Speed** 面板判断当前调度速度是否达到 TiKV 限流设置。通过调整 TiKV Snapshot Limit ([`snap-io-max-bytes-per-sec`](/tikv-configuration-file.md#snap-io-max-bytes-per-sec)) 来增加或减少该节点的调度速度。
```

**Pull request merged**